nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,536 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.sympa;
+  user = cfg.user;
+  group = cfg.group;
+  pkg = cfg.package.override { inherit (cfg) dataDir; };
+  dataDir = cfg.dataDir;
+  fqdns = mapAttrsToList (name: _val: name) cfg.domains;
+
+  sympaSubServices = [
+    "sympa-archive.service"
+    "sympa-bounce.service"
+    "sympa-bulk.service"
+    "sympa-task.service"
+  ];
+
+  sympaServiceConfig = srv: {
+    Type = "forking";
+    Restart = "always";
+    ExecStart = "${pkg}/bin/${srv}.pl";
+    PIDFile = "/run/sympa/${srv}.pid";
+    User = "${user}";
+    Group = "${group}";
+  };
+
+  mainConfig = pkgs.writeText "sympa.conf" ''
+    domain     ${if cfg.mainDomain != null then cfg.mainDomain else (head fqdns)}
+    listmaster ${concatStringsSep "," cfg.listMasters}
+    lang       ${cfg.lang}
+
+    home        ${dataDir}/list_data
+    arc_path    ${dataDir}/arc
+    bounce_path ${dataDir}/bounce
+
+    db_type ${cfg.database.type}
+    db_name ${cfg.database.name}
+    ${optionalString (cfg.database.host != null) "db_host ${cfg.database.host}"}
+    ${optionalString (cfg.database.port != null) "db_port ${cfg.database.port}"}
+    ${optionalString (cfg.database.user != null) "db_user ${cfg.database.user}"}
+    db_passwd #dbpass#
+
+    ${optionalString (cfg.mta.type == "postfix") ''
+      sendmail         /run/wrappers/bin/sendmail
```

Comment:
Probably set the `sendmail` directive always, because it's also the same path for other MTAs iirc

Reply:
+1